lab_1/main.py

```diff
@@ -87,25 +150,53 @@ def get_adjacent_words(tokens: list, word: str, left_n: int, right_n: int) -> li
     right_n = 3
     --> [['man', 'is'], ['dog, 'cat']]
     """
-    pass
+
+    check = [isinstance(tokens, list),
+             isinstance(word, str),
+             isinstance(left_n, int),
+             isinstance(right_n, int),
+             not isinstance(left_n, bool),
+             not isinstance(right_n, bool)]
+    if not all(check):
+        return []
+
+    concordance = get_concordance(tokens, word, left_n, right_n)
+    window = (left_n, right_n)
+
+    if window[0] > 0 and window[1] > 0:
+        return [[token[0], token[-1]] for token in concordance]
+    if window[0] > 0:
+        return [[token[0]] for token in concordance]
+    if window[1] > 0:
+        return [[token[-1]] for token in concordance]
+    else:
+        return []
 
 
 def read_from_file(path_to_file: str) -> str:
     """
     Opens the file and reads its content
     :return: the initial text in string format
     """
-    with open(path_to_file, 'r', encoding='utf-8') as fs:
-        data = fs.read()
+    if isinstance(path_to_file, str):
+        return ' '
 
+    with open('C:\\projects\2020-2-level-labs\\lab_1\\data.txt', 'r', encoding='utf-8') as file:
+        data = file.read()
```

Comment:
You should not have changed this function - please, return what was here.

This path is correct only for your computer and not for others.